boards/arm/nrf52840dongle_nrf52840/Kconfig

```diff
@@ -22,8 +22,8 @@ config BOARD_HAS_NRF5_BOOTLOADER
 	  If selected, applications are linked so that they can be loaded by Nordic
 	  nRF5 bootloader.
 
-config BOARD_SERIAL_BACKEND_CDC_ACM
```

Comment:
Maybe the application/sample could simply explicitly disable the option for this board if it's not intended to be used (that's what we do in nRF Desktop application)? Allowing applications to disable USB CDC ACM logging with a single Kconfig option is quite convenient approach. Since the same option name may be shared among boards (Thingy:53 already uses it), the application may even disable it in globally the application's Kconfig definition file (overlay Kconfig option and set the default value to `n`).

Alternatively, maybe we could keep the option as it was, but simply avoid enabling it when USB Bluetooth Device Class support is enabled (add only the `default y if !USB_DEVICE_BLUETOOTH`)?

btw. the PR seems to enable USB stack by default (even if we disable the `CONFIG_BOARD_NRF52840DONGLE_NRF52840_SERIAL_BACKEND_CDC_ACM`, the `CONFIG_USB_DEVICE_STACK` would still be enabled). That may not be intended in most of the cases I think.

Reply:
The problem with this is depends on and select in Kconfig, the way things in zephyr use this is a frankly a mess, for example USB CDC has a `default y` and has selects and depends, this means if you attempt to touch or use any of that with other symbols which are its parent, you get a dependency loop and you can't use it. So the only way you can do *some* level of checks is to have some options forced on with the defconfig instead unfortunately. @tejlmand can comment on these problems.

Would someone using a USB dongle really want to have USB disabled for the majority of the time?

Reply:
> Would someone using a USB dongle really want to have USB disabled for the majority of the time?

Sounds inconvenient at first, but we do have snippets now so maybe it's not unreasonable.

Reply:
I had a try at including a snippet by the board but naturally you can't do that because by the time the board files are processed, the application has already been configured, so there is no way to get this to reasonably work without e.g. always assuming USB will be enabled.